hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/ConcentricCylindersGeometry.h
+ * \brief Definition of the MPCD slit channel geometry
+ */
+
+#ifndef MPCD_CONCENTRIC_CYLINDERS_GEOMETRY_H_
+#define MPCD_CONCENTRIC_CYLINDERS_GEOMETRY_H_
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+
+#ifdef __HIPCC__
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // __HIPCC__
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Concentric cylinders geometry
+
+class __attribute__((visibility("default"))) ConcentricCylindersGeometry
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param R0 Inner radius of the cylinder
+     * \param R1 Outer radius of the cylinder
+     * \param angular_speed Angular speed of the outer cylinder
+     * \param no_slip Boundary condition at the wall (slip or no-slip)
+     */
+    HOSTDEVICE ConcentricCylindersGeometry(Scalar R0, Scalar R1, Scalar angular_speed, bool no_slip)
+        : m_R0_sq(R0 * R0), m_R1_sq(R1 * R1), m_w(angular_speed), m_no_slip(no_slip)
+        {
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((rsq < m_R0_sq) - (rsq > m_R1_sq));
+        // exit immediately if no collision is found or particle is not moving normal to the wall
+        // (since no new collision could have occurred if there is no normal motion)
+        if (sign == 0 || vel.x * vel.x + vel.y * vel.y == Scalar(0))
```

Comment:
Since you are going to reuse this later, you should save it in a variable. Also, I would update the comment slightly because this isn't actually the normal velocity, I think?
```suggestion
        // exit immediately if no collision is found or particle is stationary in xy plane
        const Scalar vxy_sq = vel.x * vel.x + vel.y * vel.y;
        if (sign == 0 || vxy_sq == Scalar(0))
```